content/concepts/did-ddo.md

```diff
@@ -1,173 +1,521 @@
+}
+```
 
-Each asset has a `metadata` service and at least one other service.
+### Services
 
-Each service is distinguished by the `DDO.service.type` attribute.
+Services define the access to the asset.
 
-Each service has an `attributes` section holding the information related to the service. That section _must_ have a `main` sub-section, holding all the mandatory information that a service has to provide.
+| Attribute              | Type                        | Required                        | Description                                                                                                                                |
+| ---------------------- | --------------------------- | ------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| **`type`**             | `string`                    | **✓**                           | Type of service (`access`, `compute`, `wss`, etc.                                                                                          |
+| **`name`**             | `string`                    |                                 | Service friendly name                                                                                                                      |
+| **`description`**      | `string`                    |                                 | Service description                                                                                                                        |
+| **`datatokenAddress`** | `string`                    | **✓**                           | Datatoken address                                                                                                                          |
+| **`providerEndpoint`** | `string`                    | **✓**                           | Provider endpoint URI (URI + path)                                                                                                         |
+| **`timeout`**          | `number`                    | **✓**                           | describing how long the service can be used after consumption is initiated. A timeout of 0 represents no time limit. Expressed in seconds. |
+| **`files`**            | Array of [Files](#files)    | **✓**                           | Array of [Files](#files) objects for publishing. These will be transformed to `encryptedFiles` during publish process.                     |
```

Comment:
`Array of Files objects for publishing. These will be transformed to encryptedFiles during publish process.` - this phrasing should not be here. This is the structure of the ddo, there is no more converting/transforming here. In this place we don't care how you handle publishing .